### DISCUSSION TITLE
Base class for object identification in PorePy

### DISCUSSION THREAD

**Discussion by vlipovac** [pmgbergen/porepy#945] [Ideas]
Hei,

I see various ways of object/instance identification implemented and used throughout PorePy.

Have you ever considered introducing a base class as a layer between Python's `Object` and whatever follows next in the package for doing that? (Grid, ADOperator, Model, Discretization,......)
I.e. a class which generates some consecutive numbers which serve as unique IDs for objects created in pp session or runscript.
Additionally a name tag can be implemented as well.
Such a class would also open possibilities to introduce further management options of created instances, if necessary.

**Comment by keileg** [comment]
Not really. We have IDs for some classes (notably grids, ad variables) and may introduce this in other places on demand, but we have never felt the need to do this in a more general setting.
